compiler/x/codegen/OMRX86Instruction.cpp

```diff
@@ -3923,6 +3923,31 @@ TR::AMD64RegImm64SymInstruction::autoSetReloKind()
 // Generate methods
 ////////////////////////////////////////////////////////////////////////////////
 
+TR::Instruction* generateDFCheck(TR::CodeGenerator *cg, TR::Instruction* cursor)
+   {
+   static char *doDFCheck = feGetEnv("TR_doDFCheck");
+   if (doDFCheck)
```

Comment:
Why is this check on the interior of this function rather than in the code that calls it?  Perhaps this comes down to how you expect this function to be used.  Is the intent that calls to this function will be unconditional and whether or not to actually emit the check will be decided here?  If calls will be conditional, why not move this check to be part of that "outer" conditional guarding it?

Reply:
I put the check inside this function because I want to call the function unconditionally in Openj9 and let the environment variable to decide whether to emit the check. But yes, this can be moved to the code that calls this function and let the caller to decide whether they want to emit the check conditionally or unconditionally.